src/actions.js

```diff
@@ -135,50 +139,123 @@ export const clearInitWallet = () => ({ type: types.SET_INIT_WALLET, payload: nu
+ *
+ * wallet {HathorWallet | HathorWalletServiceWallet} wallet object
+ * token {string} Token uid
+ * history {Array} current token history array
+ */
+export const fetchMoreHistory = async (wallet, token, history) => {
+  const newHistory = await wallet.getTxHistory({ token_id: token, skip: history.length });
+  const newHistoryObjects = newHistory.map((element) => mapTokenHistory(element, token));
+
+  return newHistoryObjects;
+};
+
 export const startWallet = (words, pin) => (dispatch) => {
   // If we've lost redux data, we could not properly stop the wallet object
   // then we don't know if we've cleaned up the wallet data in the storage
   walletUtil.cleanLoadedData();
 
-  const connection = new Connection({
-    network: 'mainnet', // app currently connects only to mainnet
-    servers: ['https://mobile.wallet.hathor.network/v1a/'],
-  });
+  const networkName = 'testnet';
 
-  const beforeReloadCallback = () => {
-    dispatch(activateFetchHistory());
-  };
+  let wallet;
+  // TODO substitute for feature flag when rollout code is implemented
+  if (false) {
+    const connection = new Connection({
+      network: networkName, // app currently connects only to mainnet
+      servers: ['https://node1.foxtrot.testnet.hathor.network/v1a/'],
+    });
 
-  const walletConfig = {
-    seed: words,
-    store: STORE,
-    connection,
-    beforeReloadCallback
-  };
+    const beforeReloadCallback = () => {
+      dispatch(activateFetchHistory());
+    };
 
-  const wallet = new HathorWallet(walletConfig);
+    const walletConfig = {
+      seed: words,
+      store: STORE,
+      connection,
+      beforeReloadCallback
+    };
+
+    wallet = new HathorWallet(walletConfig);
+  } else {
+    const network = new Network(networkName);
+    wallet = new HathorWalletServiceWallet(words, network);
+  }
 
   dispatch(setWallet(wallet));
 
   dispatch(fetchHistoryBegin());
 
+  wallet.on('state', (state) => {
+    if (state === HathorWallet.ERROR) {
+      // ERROR
+      dispatch(fetchHistoryError());
+    } else if (state === HathorWallet.READY || state === 'Ready') {
+      // READY
+      fetchHistoryAndBalance(wallet).then((data) => {
+        dispatch(fetchHistorySuccess(data));
+      });
+    }
+  });
+
   wallet.start({ pinCode: pin, password: pin }).then((serverInfo) => {
-    dispatch(setServerInfo(serverInfo));
-    wallet.on('state', (state) => {
-      if (state === HathorWallet.ERROR) {
-        // ERROR
-        dispatch(fetchHistoryError());
-      } else if (state === HathorWallet.READY) {
-        // READY
-        const historyTransactions = wallet.getTxHistory();
-        dispatch(fetchHistorySuccess(historyTransactions));
-      }
-    });
+    walletUtil.storePasswordHash(pin);
+    walletUtil.storeEncryptedWords(words, pin);
+    dispatch(setServerInfo({ version: null, network: networkName }));
 
-    wallet.on('new-tx', (tx) => {
+    // XXX no real time update for now
+    /* wallet.on('new-tx', (tx) => {
```

Comment:
This is removing the real time update for both modes (full node API and wallet service). Shouldn't we keep it enabled for the full node API?

Reply:
The method that was called when a new tx arrived changed a bit and I am not sure how the update will happen in the wallet service facade. That's why I preferred not to waste time making the real time work for the old facade because it might not work for the new one also and end up being refactored. 

I don't think we should handle this in this PR.

Reply:
I don't think we should merge it in `dev` unless the full node's Wallet API is working because it will block future rollouts of the wallet. I mean, we cannot release a new version without the real time update for the full node's Wallet API.

Reply:
I agree. I will try to make the real time work in the full node facade in a way that we lose the minimum work as possible when we develop the new real time handling.

Reply:
Real time working now for old facade. https://github.com/HathorNetwork/hathor-wallet-mobile/pull/111/commits/4ea14e548db702a271e092250251e5062394b09f